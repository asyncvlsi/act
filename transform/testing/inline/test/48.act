function addone (int x) : int
{ 
  chp {
    self := x + 1
  }
}


defproc test ()
{
  int x[4];
  int y;

  chp {
    x[addone(y)] := addone (y)
  }
}

test t;
