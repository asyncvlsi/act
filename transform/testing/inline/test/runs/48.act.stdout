defproc test ();
function addone (int<32> x) : int<32>;

defproc test ()
{

/* instances */
int<32> y;
int<32> x[4];

/* connections */
chp {
x[int(int(y,32)+1,32)]:=int(int(y,32)+0x1,32)
}
}

function addone (int<32> x) : int<32>
{

/* instances */

/* connections */
chp {
self:=x+0x1
}
}


/* instances */
test t;

/* connections */
